src/main/java/net/dv8tion/jda/api/utils/AttachmentProxy.java

```diff
@@ -259,4 +259,47 @@ public CompletableFuture<Icon> downloadAsIcon(int width, int height)
     {
         return downloadAsIcon(getUrl(width, height));
     }
+
+    /**
+     * Returns a {@link FileUpload} which supplies a data stream of this attachment,
+     * with the given file name and at the specified size.
+     * <br>The returned {@link FileUpload} can be reused safely, and does not need to be closed.
+     *
+     * <p>The attachment, if an image, may be resized at any size, however if the size does not fit the ratio of the image, then it will be cropped as to fit the target size.
+     * <br>If the attachment is not an image then the size parameters are ignored and the file is downloaded.
+     *
+     * @param  name
+     *         The name of the to-be-uploaded file
+     * @param  width
+     *         The width of this image, must be positive
+     * @param  height
+     *         The height of this image, must be positive
+     *
+     * @throws IllegalArgumentException
+     *         If any of the follow checks are true
+     *         <ul>
+     *             <li>The file name is null or blank</li>
+     *             <li>The requested width is negative or 0</li>
+     *             <li>The requested height is negative or 0</li>
+     *         </ul>
+     *
+     * @return {@link FileUpload} from this attachment.
+     */
+    @Nonnull
+    public FileUpload downloadAsFileUpload(@Nonnull String name, int width, int height)
```

Comment:
We should probably add an overload that just uses the known attachment filename

Reply:
We don't have the file name as `AttachmentProxy` is used by `MessageEmbed.Thumbnail` and `MessageEmbed.ImageInfo` and they don't have that info